### DISCUSSION TITLE
Add nginx proxy manager to the stack

### DISCUSSION THREAD

**Discussion by github-ccnt-nib** [noxPHX/Power-Hole#4] [Ideas]
Hi,
I'm currently running pi-hole and nginx-proxy manager plus bind on several VMs managing 3 domains, their subdomains, dyndns and proxying to wordpress, lamp stacks, matrix homeserver and other web-based backends. 
In order to streamline my setup a bit and (mainly) to improve the local name resolution (which currently sucks, using pihole as resolver with manual entries for local DNS) I'd be willing to look into powerDNS, so I found this project. 
I'd love to give it a try, but I can't ditch nginx-proxy-manager, which allows me to whip up a new (sub)domain in a few minutes, completely ready with letsencrypt cert and lamp server (or any other backend). 

Hence my question: Is it feasible to replace your nginx container with nginx proxy manager? 
Or can one short-circuit the embedded nginx? 
Or do you want to give it a shot? Maybe it'll add more oomp for almost no effort to your stack. 

nginx proxy manager webpage: https://nginxproxymanager.com/

Thanks for consideration.

Cheers, nib.

**Comment by noxPHX** [comment]
Hello,
If you already have a reverse proxy you want to use, I guess you can just remove the nginx service from the compose file (Or start all the services except this one).
Be sure to correctly setup the configuration on your nginx proxy manager and make sure PiHole and PowerDNS admin are reachable from NPM.
You can adjust the declared private networks or make both the services binded on the host.
Cheers

**Comment by github-ccnt-nib** [comment]
Thanks. Well, the nginx proxy manager is basically an nginx with a gui wrapped around it for user friendly configuration and certificate management. It should be possible to use it as a drop in replacement. 

I’ll start to try as you suggest and if this works, I may attempt to do the replacement. 

Cheers, nib
